File: python/perso/m2i_auto_sign.py

# Découverte Selenium sur temps personnel
""" programme pour signer automatiquement la feuille de présence, 
matin et après-midi durant la formation. Désormais non fonctionnel 
car données personnelles masquées et date actuelle non utilisable sur la feuille. """ 


from selenium import webdriver
from selenium.webdriver.common.by import By
import datetime, time

# Définir une fonction qui retourne date du jour + matin ou après-midi :
# (+ formatage de la variable pour l'utiliser en css_selector)
def date_jour_css_format():

    info_jour = datetime.datetime.now()
    date = datetime.datetime.strftime(info_jour, "\\3%m\/%d\/%Y")
    heure = "am" if datetime.datetime.strftime(info_jour, "%H") < "13" else "pm"
    
    return date[:3] + " " + date[3:] + heure

def signature_auto(navigateur) :
    navigateur.maximize_window()
    navigateur.implicitly_wait(30)
    navigateur.get('https://sign.m2iformation.fr')

    # Cliquer sur "Espace stagiaire" :
    lien_1 = navigateur.find_element(By.CSS_SELECTOR, ".col-md-8 > div:nth-child(1) > div:nth-child(2) > div:nth-child(1) > a:nth-child(1) > h5:nth-child(1)")
    lien_1.click()
    time.sleep(1)

    # Remplir le champ "Email ou n° de télephone" avec l'identifiant :
    champ_1 = navigateur.find_element(By.CSS_SELECTOR, "#inputPhoneNumber")
    champ_1.send_keys("**********")

    # Remplir le champ "Code pin" :
    champ_2 = navigateur.find_element(By.CSS_SELECTOR, "#inputSmsCode")
    champ_2.send_keys("*****")

    # Cliquer sur le bouton "Se connecter" :
    bouton_1 = navigateur.find_element(By.CSS_SELECTOR, "#connexion")
    bouton_1.click()

    # Cliquer sur le lien "Feuille de présence" :
    lien_2 = navigateur.find_element(By.CSS_SELECTOR, ".main_nav > li:nth-child(2) > a:nth-child(1) > span:nth-child(1)")
    #ActionChains(navigateur).move_to_element(lien_2).click(lien_2)
    lien_2.click()
    time.sleep(2)

    # Cliquer sur l'image "Emarger ici" correspondant à la date du jour :
    img_1 = navigateur.find_element(By.CSS_SELECTOR, f"#{date_jour_css_format()}")
    img_1.click()
    time.sleep(2)

    # Retour en haut de la page :
    lien_2 = navigateur.find_element(By.CSS_SELECTOR, ".main_nav > li:nth-child(2) > a:nth-child(1) > span:nth-child(1)")
    lien_2.click()

    # Cliquer sur "Valider et se déconnecter" :
    lien_3 = navigateur.find_element(By.CSS_SELECTOR, ".login_button > a:nth-child(1)")
    lien_3.click()
    time.sleep(1)

    navigateur.quit()

if __name__ == "__main__" :
    signature_auto(webdriver.Chrome())

File: selenium/src-test/java/m2i_auto_sign.java
/*
1er exercice : automatiser la signature de la feuille de feuille de présence
(Transposition du programme m2i_auto_sign depuis la version python)
Désormais non fonctionnel car données personnelles masquées
et date actuelle non utilisable sur la feuille.
*/

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.chrome.ChromeDriver;
// import org.openqa.selenium.firefox.FirefoxDriver;

import java.text.DateFormat;
import java.text.SimpleDateFormat;
import java.util.Date;
import java.util.concurrent.TimeUnit;

public class m2i_auto_sign {
    static WebDriver driver;
    public static String DateJour() {
        Date date_jour = new Date();
        DateFormat dateFormat = new SimpleDateFormat("\\3MM\\/dd\\/yyyy");
        DateFormat heureFormat = new SimpleDateFormat("a");
        String date = dateFormat.format(date_jour);
        String heure = heureFormat.format(date_jour).toLowerCase();
        return ("#" + date.substring(0, 3) + " " + date.substring(3) + heure);
    }

    public static void main(String[] args) throws InterruptedException {

        // System.setProperty("webdriver.gecko.driver", "C:\\maven\\drivers\\geckodriver.exe");
        System.setProperty("webdriver.chrome.driver", "C:\\maven\\drivers\\chromedriver.exe");
        driver = new ChromeDriver();
        driver.get("https://sign.m2iformation.fr");
        driver.manage().window().maximize();
        driver.manage().timeouts().implicitlyWait(30, TimeUnit.SECONDS);

        // Cliquer sur "Espace stagiaire"
        driver.findElement(By.cssSelector(".col-md-8 > div:nth-child(1) > div:nth-child(2) > div:nth-child(1) > a:nth-child(1) > h5:nth-child(1)")).click();

        // Remplir le champ "Email ou n° de télephone" avec l'identifiant :
        driver.findElement(By.cssSelector("#inputPhoneNumber")).sendKeys("**********");

        // Remplir le champ "Code pin" :
        driver.findElement(By.cssSelector("#inputSmsCode")).sendKeys("*****");

        // Cliquer sur le bouton "Se connecter" :
        driver.findElement(By.cssSelector("#connexion")).click();
        //TimeUnit.SECONDS.sleep(5);

        // Cliquer sur le lien "Feuille de présence" :
        driver.findElement(By.cssSelector(".main_nav > li:nth-child(2) > a")).click();
        TimeUnit.SECONDS.sleep(1);

        // Cliquer sur l'image "Emarger ici" correspondant à la date du jour :
        driver.findElement(By.cssSelector(DateJour())).click();
        TimeUnit.SECONDS.sleep(1);

        // Retour en haut de la page :
        driver.findElement(By.cssSelector(".main_nav > li:nth-child(2) > a:nth-child(1) > span:nth-child(1)")).click();

        // Cliquer sur "Valider et se déconnecter" :
        driver.findElement(By.cssSelector(".login_button > a")).click();

        TimeUnit.SECONDS.sleep(1);
        driver.close();
    }
}